lib/customer.rb

```diff
@@ -18,4 +18,12 @@ def invoices
   def find_all_invoices_by_customer_id(id)
     parent.find_all_invoices_by_customer_id(id)
   end
+
+  def transactions
+   invoices.flat_map(&:transactions)
+  end
+
+  def favorite_merchant
+    invoices.flat_map(&:merchants).max_by { |invoices| invoices(&:successful_charge?) }
```

Comment:
Unused block argument - `invoices`. You can omit the argument if you don't care about it.<br>Line is too long. [87/80]